### PR TITLE
Revised WIFI scanning, connection, output to serial log instead of screen. 

### PR DESCRIPTION
**Pull request by zhenwei2972**
automatically connects to test WIFI network, does not crash anymore. 
ID: Test
Password: password

### CODE REVIEW

File: alpha/hal.cpp [hunk 1]
```diff
@@ -324,7 +328,47 @@ void _TS_HAL::log_init()
 #endif
 }
 
-
+//
+// WIFI setup
+// 
+void _TS_HAL::wifi_init()
```
Comment (jamesi8086) [inline alpha/hal.cpp]: Lets put all the wifi things into a file called radio.h/cpp.
Eventually BT things should move there as well.

wifi should be able to turn on/off on-demand, no  need to init in setup()

File: alpha/hal.cpp [hunk 2]
```diff
@@ -324,7 +328,47 @@ void _TS_HAL::log_init()
...
+  	n = WiFi.scanNetworks();
+  	if (n > 3)
+  	{
+  		n = 3;
+  	}
+  	for (int i = 0; i < n; ++i)
+  	{
+  		//prints top 3 nearest WIFI SSIDS
+  		log(WiFi.SSID(i));
+  		delay(1000);
```
Comment (jamesi8086) [inline alpha/hal.cpp]: use HAL delay function

Comment (jamesi8086) [reply alpha/hal.cpp]: or no delays at all. convert this whole logic into smaller functions to allow polling, or use event callbacks if they exist.

File: alpha/hal.cpp [hunk 3]
```diff
@@ -324,7 +328,47 @@ void _TS_HAL::log_init()
...
+  
+  	if (n == 0)
+  	{
+  		log("no wifi found \n");
+  	}
+  	else
+  	{
+  		//Connect to, prints name of connected WIFI. 
+  		WiFi.begin(WIFI_SSID, WIFI_PASS);
+      while (WiFi.status() != WL_CONNECTED)
```
Comment (jamesi8086) [inline alpha/hal.cpp]: no while loops/delays here.

Comment (jamesi8086) [reply alpha/hal.cpp]: convert this whole logic into smaller functions to allow polling, or use event callbacks if they exist.

File: alpha/radio.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+// WIFI setup
+// 
+void _TS_RADIO::wifi_connect()
+{
+  if (!this->wifiInitialized)
+  {
+    WiFi.mode(WIFI_STA);
+    WiFi.disconnect();
+    int SSID_COUNT = 0;
+    SSID_COUNT = WiFi.scanNetworks();
```
Comment (jamesi8086) [inline alpha/radio.cpp:44]: I don't understand, whats the purpose of this?

Should it look for WIFI_SSID and only connect if it exists/visible?

How long does scanNetworks take?

File: alpha/alpha.ino [hunk 5]
```diff
@@ -12,7 +13,8 @@ bool powerSaveTest = false;
 void setup() {
   TS_HAL.begin();
   TS_HAL.ble_init();
-  TS_HAL.wifi_init();
+  TS_RADIO.begin();
+  TS_RADIO.wifi_connect();
```
Comment (jamesi8086) [inline alpha/alpha.ino]: lets not connect on setup. don't add into main file until its tested to be compatible with the main radio loop, which would only be brought in once @fustilio figures out the power state machine.

Comment (jamesi8086) [reply alpha/alpha.ino]: e.g. 
powered by wire: enable wifi and tasks
powered by battery: disable wifi

File: alpha/hal.cpp [hunk 6]
```diff
@@ -10,9 +10,9 @@
 #include <WiFi.h>
 #define WIFI_SSID "Test"       // Enter your SSID here
```
Comment (jamesi8086) [inline alpha/hal.cpp]: all these should be in radio.h/cpp ?

File: alpha/hal.cpp [hunk 7]
```diff
@@ -331,38 +330,41 @@ void _TS_HAL::log_init()
 //
 // WIFI setup
 // 
-void _TS_HAL::wifi_init()
+void _TS_HAL::wifi_connect()
```
Comment (jamesi8086) [inline alpha/hal.cpp]: why does code here look copied to radio.cpp

File: alpha/hal.cpp (hunk 6 above)
Comment (zhenwei2972) [reply alpha/hal.cpp]: Yep will move it

File: alpha/hal.cpp (hunk 7 above)
Comment (zhenwei2972) [reply alpha/hal.cpp]: I shifted it over to radio.cpp because we wanted to seperate it ? Not sure if I misunderstood

File: alpha/radio.cpp (hunk 4 above)
Comment (zhenwei2972) [reply alpha/radio.cpp:44]: wifi_connect primarily is meant to find available networks, then connect to a preset one, can be simplifed.
It includes printing to console to display which SSIDs are available, when not available, to print an error.

scanNetworks is almost instant, but I found that having at least a one second wait gives it some buffer to be more reliable

File: alpha/alpha.ino (hunk 5 above)
Comment (zhenwei2972) [reply alpha/alpha.ino]: Ok will remove that call in main

**Pull request merged**